marshmallow/fields.py

```diff
@@ -915,7 +916,6 @@ def _deserialize(self, value, attr, data):
         """Deserialize an ISO8601-formatted time to a :class:`datetime.time` object."""
         if not value:   # falsy values are invalid
             self.fail('invalid')
-            raise err
```

Comment:
I can't merge this PR, but I would encourage keeping PRs as concise as possible for the sake of the reviewer.

It is my personal preference to keep unrelated changes out of a PR, although it is generally acceptable to include necissary formatting fixes in files you modified in a separate commit.
http://stackoverflow.com/questions/2214480/committing-when-changing-source-formatting

Removing unrelated lines of code without testing the behavior is probably not a good idea. Opening a separate PR would help keep unrelated changes from getting blocked on each other.